sdk/storage/azblob/client_test.go

```diff
@@ -335,6 +337,78 @@ func (s *AZBlobUnrecordedTestsSuite) TestUploadAndDownloadFileNonZeroOffsetAndCo
 	performUploadAndDownloadFileTest(s.T(), _require, testName, fileSize, blockSize, concurrency, downloadOffset, downloadCount)
 }
 
+func (s *AZBlobUnrecordedTestsSuite) TestMD5SmallFileUpload() {
+	_require := require.New(s.T())
+	fileSize := 1 * 1024 * 1024 // 1 MB file
+
+	md5sum, resp, err := performUploadWithMD5Check(s.T(), _require, s.T().Name(), fileSize)
+
+	_require.NoError(err)
+
+	md5sumStr := hex.EncodeToString(md5sum)
+	respMd5 := hex.EncodeToString(resp.ContentMD5)
+	_require.Equal(md5sumStr, respMd5)
+
+}
+
+func (s *AZBlobUnrecordedTestsSuite) TestMD5LargeFileUpload() {
+	_require := require.New(s.T())
+	fileSize := 257 * 1024 * 1024 // 257 MB file, 256 MB is the max for calculate md5 and service md5 won't match
+
+	md5sum, resp, err := performUploadWithMD5Check(s.T(), _require, s.T().Name(), fileSize)
+
+	_require.ErrorContains(err, "400 The MD5 value specified in the request did not match with the MD5 value calculated by the server.")
```

Comment:
Why are we getting this error for large file uploads?

Reply:
a transactional md5 will always be calculated by the service. If the service is failing the transactional md5 check it's because the check is actually failing. I imagine this is because you've configured the transactional md5 on put blocklist. A transactional md5 for that API is literally checking the content integrity of the blocklist itself (it's one of the only places .NET SDK currently doesn't calculate one).

Transactional md5 validates individual requests and nothing more, meaning it would need to validate individual blocks. If one large md5 is being passed in for a partitioned upload, the client should fail fast since the checksum cannot be split.